docs/dotnet/dotnet-http-metrics.md

```diff
@@ -0,0 +1,238 @@
+- [HTTP server](#http-server)
+  * [Metric: `http.server.request.duration`](#metric-httpserverrequestduration)
+  * [Metric: `http.server.active_requests`](#metric-httpserveractive_requests)
+
+<!-- tocstop -->
+
+## HTTP client
+
+All Http client metrics are reported by the `System.Net.Http` meter.
+
+### Metric: `http.client.request.duration`
+
+Client request duration measures the time it takes to receive response headers and doesn't include time to read the response body.
+
+This metric follows the common [http.client.request.duration](../http/http-metrics.md#metric-httpclientrequestduration) definition.
+
+Notes:
+
+- Meter name is `System.Net.Http`
+- Metric added in .NET 8.0
+- When the `error.type` attribute is reported, it contains one of [HTTP Request errors](https://learn.microsoft.com/dotnet/api/system.net.http.httprequesterror) in snake_case, a full exception type, or a string representation of received status code.
+
+### Metric: `http.client.open_connections`
+
+<!-- semconv metric.dotnet.http.client.open_connections(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `http.client.open_connections` | UpDownCounter | `{connection}` | Number of outbound HTTP connections that are currently active or idle on the client [1] |
+
+**[1]:** Meter name: `System.Net.Http`; Added in: .NET 8.0
+<!-- endsemconv -->
+
+<!-- semconv metric.dotnet.http.client.open_connections(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `http.connection.state` | string | State of the HTTP connection in the HTTP connection pool. | `active`; `idle` | Required |
+| [`network.peer.address`](../attributes-registry/network.md) | string | Remote IP address of the socket connection. | `10.1.2.80` | Recommended |
+| [`network.protocol.version`](../attributes-registry/network.md) | string | Version of the protocol specified in `network.protocol.name`. [1] | `1.1`; `2` | Recommended |
+| [`server.address`](../general/attributes.md) | string | Host identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) the HTTP request is sent to. [2] | `example.com`; `10.1.2.80`; `/tmp/my.sock` | Required |
+| [`server.port`](../general/attributes.md) | int | Port identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) the HTTP request is sent to. [3] | `80`; `8080`; `443` | Conditionally Required: [4] |
+| [`url.scheme`](../attributes-registry/url.md) | string | The [URI scheme](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) component identifying the used protocol. | `http`; `https`; `ftp` | Recommended |
+
+**[1]:** `network.protocol.version` refers to the version of the protocol used and might be different from the protocol client's version. If the HTTP client has a version of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
```

Comment:
I have hard time following the wording of the [original spec](https://github.com/open-telemetry/semantic-conventions/blob/c26aad838312eea9d53d781fb4328c2c61f9ea08/docs/attributes-registry/network.md)  this was copied from (what does "protocol client's version" mean???), but note that for connection metrics, `network.protocol.version` is being reported differently anyways. Considering [the way this is implemented](https://github.com/dotnet/runtime/blob/748dd8edecc3cc2b671cab7437adc4b0c750d0be/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionBase.cs#L47-L51), I recommend the following text, though not sure if the wording is compliant:

```suggestion
**[1]:** In connection metrics `network.protocol.version` refers to the version of the underlying connection pool and might be different from the version specified in [`HttpRequestMessage.Version`](https://learn.microsoft.com/dotnet/api/system.net.http.httprequestmessage.version?view=net-7.0) or [`HttpClient.DefaultVersion`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.defaultrequestversion?view=net-7.0). Ie. if `HttpRequestMessage.Version` has a value of `1.0`, this attribute will be set to `1.0`.
```

Reply:
I rephrased it to 

> HTTP 1.0 and 1.1 requests share connections in the connection pool and are both reported as version `1.1`. So, the `network.protocol.version` value reported on connection metrics is different than the one reported on request-level metrics or spans for HTTP 1.0 requests.

WDYT?

Reply:
Definitely sounds better than my proposal!